Bottom: 03147211e4c277726edc0f1afb8f0cde7fe7b6f2
Top:    2f95dab57b18d062043ebc4da91a2c2b51f087ec
Author: Andrew Cooper <andrew.cooper@novatechweb.com>
Date:   2018-05-17 16:47:42 -0500

Add vmdaedalus inventory

Run ansible provisioning through Docker

Add 'restore' tag to tasks. By tagging the tasks, restoration can be skipped using --skip-tags
option to ansible-playbook. This is useful for testing ansible plays
on systems without tapedrives or established backups.

Add Vagrant configuration

Set docker container restart_policy to no for testing purposes

Add bacula-restores task to symlink backup files to bacula restore folder

Comment out restore file permission tasks:
- can't set permissions on symlinks


---
diff --git a/.gitignore b/.gitignore
index e69de29b..c1a0118f 100644
--- a/.gitignore
+++ b/.gitignore
@@ -0,0 +1 @@
+/.vagrant/
diff --git a/Vagrantfile b/Vagrantfile
new file mode 100644
index 00000000..488140b6
--- /dev/null
+++ b/Vagrantfile
@@ -0,0 +1,103 @@
+$bbworker = <<-SCRIPT
+yum install -y \
+    epel-release
+
+yum install -y \
+    device-mapper-persistent-data \
+    git \
+    lvm2 \
+    python-pip \
+    yum-utils
+
+yum-config-manager \
+  --add-repo \
+  https://download.docker.com/linux/centos/docker-ce.repo
+
+yum remove -y \
+    docker \
+    docker-client \
+    docker-client-latest \
+    docker-common \
+    docker-latest \
+    docker-latest-logrotate \
+    docker-logrotate
+
+yum install -y \
+    docker-ce
+
+pip install \
+    docker
+
+systemctl start docker
+docker run --rm hello-world
+SCRIPT
+
+Vagrant.configure("2") do |config|
+
+    config.vm.define "vmdaedalus", primary: true do |main|
+        main.vm.box = "daedalus"
+        main.vm.box_url = "vmdaedalus/vmdaedalus-virtualbox.box"
+        main.vm.hostname = "vmdaedalus.novatech-llc.com"
+        main.vm.provider "virtualbox" do |vb|
+            vb.name = "vmdaedalus"
+            vb.memory = 6000
+            vb.cpus = 4
+        end
+    #    config.vm.network "public_network", ip: "192.168.0.100", netmask:"255.255.0.0", bridge: "enp0s31f6"
+        main.vm.network "private_network", 
+            ip: "192.168.0.100", 
+            netmask:"255.255.0.0"
+
+        main.vm.network "forwarded_port",
+            guest: 2375,
+            host: 12375,
+            protocol: "tcp",
+            id: "docker"
+
+        main.vm.network "forwarded_port", 
+            guest: 22, 
+            host: 10022,
+            id: "ssh"
+
+        main.vm.synced_folder "/mnt/vm/bacula-restores",
+            "/mnt/bacula-storage",
+            id: "bacula-storage",
+            owner: "root",
+            group: "root",
+            mount_options: ["ro"]
+    end
+
+    config.vm.define "bbworker" do |worker|
+        worker.vm.box = "centos/7"
+        worker.vm.hostname = "bbworker.novatech-llc.com"
+        worker.vm.provider "virtualbox" do |vb|
+            vb.name = "bbworker"
+            vb.cpus = 1
+        end
+        
+        worker.vm.network "private_network", 
+            ip: "192.168.0.150", 
+            netmask:"255.255.0.0"
+
+        worker.vm.network "forwarded_port",
+            guest: 2375,
+            host: 22375,
+            protocol: "tcp",
+            id: "docker"
+
+        worker.vm.network "forwarded_port", 
+            guest: 22, 
+            host: 20022,
+            id: "ssh"
+
+        worker.vm.provision "shell", inline: $bbworker
+
+        worker.vm.synced_folder "/mnt/vm/builder",
+            "/mnt/assets",
+            id: "builder-assets",
+            owner: "root",
+            group: "root",
+            mount_options: ["rw"]
+
+    end
+end
diff --git a/ansible-playbook/host_vars/vmdaedalus.novatech-llc.com/networking.yml b/ansible-playbook/host_vars/vmdaedalus.novatech-llc.com/networking.yml
new file mode 100644
index 00000000..ccbedf8a
--- /dev/null
+++ b/ansible-playbook/host_vars/vmdaedalus.novatech-llc.com/networking.yml
@@ -0,0 +1,18 @@
+---
+
+ansible_user: vagrant
+
+daedalus_subnet: 192.168.0.0/16
+daedalus_net_connection: "enp5s0f0"
+
+svn_ip_addr:          192.168.0.101
+gitlab_ip_addr:       192.168.0.102
+wiki_ip_addr:         192.168.0.103
+mantisbt_ip_addr:     192.168.0.104
+build_ip_addr:        192.168.0.105
+testssh_ip_addr:      192.168.0.106
+exim4_ip_addr:        192.168.0.107
+openldap_ip_addr:     192.168.0.108
+phpldapadmin_ip_addr: 192.168.0.108
+phpmyadmin_ip_addr:   192.168.0.109
+buildbot_ip_addr:     192.168.0.110
diff --git a/ansible-playbook/vmdaedalus b/ansible-playbook/vmdaedalus
new file mode 100644
index 00000000..dc2fd1e9
--- /dev/null
+++ b/ansible-playbook/vmdaedalus
@@ -0,0 +1,15 @@
+vmdaedalus.novatech-llc.com ansible_host=127.0.0.1 ansible_port=10022 ansible_user='vagrant' ansible_ssh_private_key_file='/home/coopera/Developer/testdaedalus/.vagrant/machines/vmdaedalus/virtualbox/private_key'
+buildbot-worker.novatech-llc.com ansible_host=127.0.0.1 ansible_port=20022 ansible_user='vagrant' ansible_ssh_private_key_file='/home/coopera/Developer/testdaedalus/.vagrant/machines/bbworker/virtualbox/private_key'
+
+[bacula]
+vmdaedalus.novatech-llc.com
+
+[daedalus]
+vmdaedalus.novatech-llc.com
+
+[docker-hosts]
+vmdaedalus.novatech-llc.com
+buildbot-worker.novatech-llc.com
+
+[buildbot-worker]
+buildbot-worker.novatech-llc.com
diff --git a/vmdaedalus/.gitignore b/vmdaedalus/.gitignore
new file mode 100644
index 00000000..c148b6f5
--- /dev/null
+++ b/vmdaedalus/.gitignore
@@ -0,0 +1,2 @@
+bacula-restores/
+vmdaedalus-virtualbox.box
diff --git a/vmdaedalus/vmdaedalus.cfg b/vmdaedalus/vmdaedalus.cfg
new file mode 100644
index 00000000..5d5fdee4
--- /dev/null
+++ b/vmdaedalus/vmdaedalus.cfg
@@ -0,0 +1,71 @@
+install
+text
+cdrom
+skipx
+lang en_US.UTF-8
+keyboard us
+timezone UTC
+rootpw vagrant
+user --name=vagrant --password=vagrant
+auth --enableshadow --passalgo=sha512 --kickstart
+firewall --disabled
+selinux --permissive
+bootloader --location=mbr
+zerombr
+clearpart --all --initlabel
+autopart
+firstboot --disable
+reboot
+repo --name=epel   --baseurl=http://download.fedoraproject.org/pub/epel/7/x86_64/
+repo --name=extras --baseurl=http://mirror.centos.org/centos/7/extras/x86_64/
+repo --name=docker --baseurl=http://download.docker.com/linux/centos/7/x86_64/stable/
+
+%packages --instLangs=en_US.utf8 --nobase --ignoremissing --excludedocs
+@base
+@core
+@hardware-monitoring
+@input-methods
+@performance
+@mariadb
+@mariadb-client
+epel-release
+bacula-client
+bacula-common
+bacula-console
+bacula-director
+bacula-libs
+bacula-libs-sql
+bacula-storage
+bacula-console-bat
+docker-ce
+git
+subversion
+pexpect
+yum-utils
+%end
+ 
+ 
+%post --log=/root/ks.log
+ 
+echo "vagrant ALL=(ALL) NOPASSWD: ALL" >> /etc/sudoers.d/vagrant
+echo "Defaults:vagrant !requiretty" >> /etc/sudoers.d/vagrant
+chmod 0440 /etc/sudoers.d/vagrant
+mkdir -pm 700 /home/vagrant/.ssh
+#curl -o /home/vagrant/.ssh/authorized_keys https://raw.githubusercontent.com/mitchellh/vagrant/master/keys/vagrant.pub
+cat <<EOK >/home/vagrant/.ssh/authorized_keys
+ssh-rsa AAAAB3NzaC1yc2EAAAABIwAAAQEA6NF8iallvQVp22WDkTkyrtvp9eWW6A8Y\
+Vr+kz4TjGYe7gHzIw+niNltGEFHzD8+v1I2YJ6oXevct1YeS0o9HZyN1Q9qgCgzUFtdO\
+KLv6IedplqoPkcmF0aYet2PkEDo3MlTBckFXPITAMzF8dJSIFo9D8HfdOV0IAdx4O7Pt\
+ixWKn5y2hMNG0zQPyUecp4pzC6kivAIhyfHilFR61RGL+GPXQ2MWZWFYbAGjyiYJnAmC\
+P3NOTd0jMZEnDkbUvxhMmBYSdETk1rRgm+R4LOzFUGaHqHDLKLX+FIPKcF96hrucXzcW\
+yLbIbEgE98OHlnVYCzRdK8jlqm8tehUc9c9WhQ== vagrant insecure public key
+EOK
+chmod 0600 /home/vagrant/.ssh/authorized_keys
+chown -R vagrant.vagrant /home/vagrant/.ssh
+yum -y update
+yum -y remove linux-firmware
+yum-config-manager \
+    --add-repo \
+    https://download.docker.com/linux/centos/docker-ce.repo
+
+%end
diff --git a/vmdaedalus/vmdaedalus.json b/vmdaedalus/vmdaedalus.json
new file mode 100644
index 00000000..2703db83
--- /dev/null
+++ b/vmdaedalus/vmdaedalus.json
@@ -0,0 +1,43 @@
+{
+    "variables": {
+    },
+    "builders": [{
+        "type": "virtualbox-iso",
+        "guest_os_type": "RedHat_64",
+        "iso_url": "/home/coopera/Downloads/CentOS-7-x86_64-DVD-1708/CentOS-7-x86_64-DVD-1708.iso",
+        "iso_checksum": "ec7500d4b006702af6af023b1f8f1b890b6c7ee54400bb98cef968b883cd6546",
+        "iso_checksum_type": "sha256",
+        "ssh_username": "vagrant",
+        "ssh_password": "vagrant",
+        "ssh_wait_timeout": "600s",
+        "vm_name": "vmdaedalus",
+        "http_directory": "./",
+        "boot_wait": "10s",
+        "boot_command": [
+            "<esc><wait>",
+            "linux ks=http://{{.HTTPIP}}:{{.HTTPPort}}/vmdaedalus.cfg",
+            "<enter><wait>"
+        ],
+        "shutdown_command": "echo 'vagrant' |sudo -S shutdown -P now",
+        "vboxmanage": [
+            ["modifyvm", "{{.Name}}", "--memory", "512"],
+            ["modifyvm", "{{.Name}}", "--usb", "off"],
+            ["modifyvm", "{{.Name}}", "--audio", "none"]
+        ],
+        "disk_size": "120000"
+    }],
+    "provisioners": [{
+        "type": "shell",
+        "inline": [
+            "sudo yum -y install gcc make bzip2 kernel-headers kernel-devel dkms",
+            "sudo yum clean all",
+            "sudo mount -o loop /home/vagrant/VBoxGuestAdditions.iso /mnt",
+            "sudo sh /mnt/VBoxLinuxAdditions.run",
+            "sudo umount /mnt"
+        ]
+    }],
+    "post-processors": [{
+        "type": "vagrant",
+        "output": "./vmdaedalus-{{.Provider}}.box"
+    }]
+}
